tests/benchmark_helpers.rs

```diff
@@ -0,0 +1,115 @@
+//! Simple tests for benchmark helper functions
+//! These ensure the benchmark code itself is correct
+
+/// Simple O(n*m) pattern matcher - copy from benchmarks
+fn naive_search(haystack: &[u8], needle: &[u8]) -> Option<usize> {
+    if needle.is_empty() || needle.len() > haystack.len() {
+        return None;
+    }
+    for i in 0..=haystack.len() - needle.len() {
+        if &haystack[i..i + needle.len()] == needle {
+            return Some(i);
+        }
+    }
+    None
+}
+
+/// Parse a hex string - copy from benchmarks
+fn parse_hex_pattern(s: &str) -> Result<Vec<u8>, String> {
```

Comment:
This is not the function used in the actual program, just a copy. Please try to use real functions.
Move related tests into the src/main.rs file instead.